src/config.rs

```diff
@@ -118,14 +119,30 @@ struct RawDisplayConfig {
     pub use_pager: bool,
 }
 
-#[derive(Debug, Default, Serialize, Deserialize, PartialEq, Eq)]
+/// Serde doesn't support default values yet (tracking issue:
+/// https://github.com/serde-rs/serde/issues/368), so we need to wrap DEFAULT_UPDATE_INTERVAL_HOURS
+/// in a function to be able to use #[serde(default = ...)]
+const fn default_auto_update_interval_hours() -> u64 {
+    DEFAULT_UPDATE_INTERVAL_HOURS
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
 struct RawUpdatesConfig {
     #[serde(default)]
     pub auto_update: bool,
-    #[serde(default)]
+    #[serde(default = "default_auto_update_interval_hours")]
     pub auto_update_interval_hours: u64,
 }
 
+impl Default for RawUpdatesConfig {
+    fn default() -> Self {
+        Self {
+            auto_update: false,
+            auto_update_interval_hours: DEFAULT_UPDATE_INTERVAL_HOURS,
+        }
+    }
+}
```

Comment:
If we use `#[serde(default...)]`, why do we need to manually impl `Default`? Wouldn't serde handle that for us, and delegate to the field default values when getting the default value for a `#[derive(Default)]` struct?

Reply:
I thought so too, but it doesn't appear that it's the case:

- Output of `println!("{:?}", RawUpdatesConfig::default())` with the current implementation:

```rust
RawUpdatesConfig { auto_update: false, auto_update_interval_hours: 720 }
```

- Output of `println!("{:?}", RawUpdatesConfig::default())` with `derive(Default)`:

```rust
RawUpdatesConfig { auto_update: false, auto_update_interval_hours: 0 }
```

Reply:
Ah I see. The `#[serde(default = ...)]` attribute only changes deserialization, but does not influence the `Default` implementation (which is used to seed a config).

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=387ce3b5e2c29cc869ca648cf0ee8dba

We could use a newtype with a different `Default`, then you don't need the custom serde function anymore. However, I'm not sure yet whether that's cleaner or not. What do you think?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f03be819dc29496526d3d93ec6d51187

Reply:
I think both options makes sense, I don't have a strong preference. Creating a new type here doesn't add too much complexity, but it could become messy if there are more fields like this in the future.

A third option I can think of is getting rid of the `Default` implementations altogether and initializing raw configs with `toml::from_str("")` instead of `RawConfig::new()`. This allows to unify the defaults (always use the serde defaults), but the performance may not be as good.

Reply:
Ok, let's stick to your approach for now. Should be simple enough to understand.